sql/src/test/java/org/apache/druid/sql/calcite/http/SqlResourceTest.java

```diff
@@ -671,6 +671,46 @@ public void testResourceLimitExceeded() throws Exception
     checkSqlRequestLog(false);
   }
 
+  @Test
+  public void testSubQueryWithLimit() throws Exception
+  {
+    final List<Map<String, Object>> rows = doPost(
+        new SqlQuery(
+            "SELECT COUNT(*) AS cnt FROM ( SELECT * FROM ( SELECT * FROM druid.foo LIMIT 10 ) tmpA ) tmpB",
+            null,
+            false,
+            null
+        )
+    ).rhs;
+
+    Assert.assertEquals(
+        ImmutableList.of(),
+        rows
+    );
+    checkSqlRequestLog(true);
+  }
+
+  @Test
+  public void testSubQueryWithoutLimit() throws Exception
```

Comment:
Could you move these tests into `CalciteQueryTest` please? Its `testQuery` method provides a nice way to test this stuff, including testing that it plans into an appropriate native query.

Reply:
Good Suggestion :+1: